docs/tsp/typeServerProtocol.ts

```diff
@@ -241,36 +269,35 @@ export namespace TypeServerProtocol {
+        ShutDown = 'typeServer/shutdown',
         // Request from client to get the current snapshot of the type server.
         // A snapshot is a point-in-time representation of the type server's state, including all loaded files and their types.
         // A type server should change its snapshot whenever any type it might have returned is no longer valid. Meaning types are
         // only usable for the snapshot they were returned with.
+        //
+        // Snapshots are not meant to survive any changes that would make the type server throw away its internal cache. They are merely an
+        // identifier to indicate to the client that the type server will accept requests for types from that snapshot.
         GetSnapshot = 'typeServer/getSnapshot',
         // Request to get diagnostics for a specific file.
         GetDiagnostics = 'typeServer/getDiagnostics',
         // Request to get the version of diagnostics for a specific file.
         GetDiagnosticsVersion = 'typeServer/getDiagnosticsVersion',
         // Request to get the type information for a specific node.
         GetType = 'typeServer/getType',
-        // Request to get the union types for a specific type.
+        // Request to get the collection of subtypes that make up a union type.
         GetUnionTypes = 'typeServer/getUnionTypes',
-        // Request to find a member of a type.
-        GetTypeMember = 'typeServer/getTypeMember',
-        // Request to get all overloads of a function or method.
+        // Request to find an attribute of a class.
+        GetTypeAttribute = 'typeServer/getTypeAttribute',
+        // Request to get all overloads of a function or method. The returned value doesn't include the implementation signature.
         GetOverloads = 'typeServer/getOverloads',
-        // Request to get the effective meta class of a type.
-        GetEffectiveMetaClass = 'typeServer/getEffectiveMetaClass',
+        // Request to get the meta class of a type.
+        GetMetaclass = 'typeServer/getMetaclass',
         // Request to get the type of a declaration.
         GetTypeOfDeclaration = 'typeServer/getTypeOfDeclaration',
         // Request to get symbol declaration information for a node.
         GetSymbolDeclarationInfo = 'typeServer/getSymbolDeclarationInfo',
-        // Request to get the bound magic method for a type.
-        GetBoundMagicMethod = 'typeServer/getBoundMagicMethod',
-        // Request to resolve an alias declaration.
-        // This request is sent by the client to resolve an alias declaration, which may involve looking up the
-        // declaration in the type server's cache or performing additional resolution logic.
-        ResolveAliasDeclaration = 'typeServer/resolveAliasDeclaration',
+        // Request to resolve an import declaration. Example: `from module import something`. The `something` is the import declaration.
+        ResolveImportDeclaration = 'typeServer/resolveImportDeclaration',
```

Comment:
I don't fully understand why this is necessary

Reply:
The declaration in this case points to the symbol 'something'. That symbol isn't defined in the current module. This function goes and finds the symbol in the other module.

Here's the location of this function in the pyright source:

https://github.com/microsoft/pyright/blob/5021af8f0545386b99e916b764bb8968d362bd7f/packages/pyright-internal/src/analyzer/declarationUtils.ts#L231

it's basically finding the original symbol.

Maybe the name of this request should be findOriginalDecl or something like that instead?